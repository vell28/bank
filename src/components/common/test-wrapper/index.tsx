import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { mainDarkTheme } from 'modules/themes';
import { store } from 'modules/store';

interface ITestWrapper {
  children: React.ReactNode;
}

export const TestWrapper: React.FC<ITestWrapper> = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={mainDarkTheme}>
      <>{children}</>
    </ThemeProvider>
  </Provider>
);
