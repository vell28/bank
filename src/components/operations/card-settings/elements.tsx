import styled from 'styled-components';

export const UnblockBox = styled.div`
  .popup-description {
    margin-bottom: 80px;
    padding-top: 20px;
  }
`;

export const CardSettingBox = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  padding: 15px 57px 24px;
  border-radius: 25px;
  background-color: ${(props) => props.theme.colors.white};
`;

export const CardDataPreviewBox = styled.div`
  width: 100%;
  padding-top: 12px;
`;

export const BlockCardBtn = styled.div`
  position: relative;
  padding: 0 29px;
  display: inline-block;
  color: ${(props) => props.theme.colors.blue};
  font-size: 16px;
  letter-spacing: 1.8px;
  transition: all 0.3s;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
`;

export const BtnsBox = styled.div`
  position: relative;
  margin-top: 36px;
  text-align: center;
`;
