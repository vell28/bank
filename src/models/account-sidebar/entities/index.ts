export interface IAccountSidebar {
  [key: string]: boolean;
}
