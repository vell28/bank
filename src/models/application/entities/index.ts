export interface IApplication {
  applicationId: string;
  productId: string;
  Authorization: string;
  deviceId?: string;

  readonly bankName: string;
  readonly BIC: string;
}
