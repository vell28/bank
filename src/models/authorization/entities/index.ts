export interface IAuthorization {
  service_code?: string;
  phone?: string;
  token?: string;
  sessid?: string;
  token_expires?: Date;
  // maybe its not required
  'dev-id'?: string;
  'token-firebase'?: string;
  isShownLogoutConfirmModal: boolean;
}

export interface ITokenRefresh {
  success: boolean;
  token: string;
}

export interface ISuccessLogout {
  success: boolean;
}

export interface ICodeRequest {
  sessid: string;
  code: string | null;
  success?: boolean;
}

export interface IPasswordCheckRequest {
  password: string;
  phone: string;
}

export interface ISuccessCheckResponse {
  status: 'ok';
}
