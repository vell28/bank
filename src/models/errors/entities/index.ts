export type AppErrorType = '401' | '404' | '403' | '500' | 'ERR_INTERNET_DISCONNECTED';

export interface IAppErrors {
  title?: string;
  subtitle?: string;
  onOk?: () => void;
  btnText?: string;
  supportUrl?: string;
  type?: AppErrorType;
  meta?: string;
}
