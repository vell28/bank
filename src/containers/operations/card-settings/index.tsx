import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { propOr } from 'ramda';
import { useTranslation } from 'react-i18next';

import {
  CardSettingBox,
  CardDataPreviewBox,
  BlockCardBtn,
  BtnsBox,
} from 'components/operations/card-settings/elements';
import { CardPreview } from 'components/common/card-preview';
import { Skeleton } from 'components/skeleton';
import { UnmaskedCardPreview } from 'components/common/unmasked-card-data';

import { AccountStatus } from 'models/organizations/entities';
import { BlockUnmaskedCardSkelet } from 'components/skeleton/collections';
import { SpinnerWrapper } from '@components/spinner';
import { PopupBlockCard } from 'components/common/popup/popup-dialog';
import {
  getFullUnmaskedSelectedCard,
  isInitialLoadingCard,
  getSliderActiveCardId,
  getSliderActiveCard,
  getIsShownLockModal,
} from 'models/operations/card-settings/redux/selectors';
import {
  toggleBlockCardModal,
  toggleUnBlockCardModal,
  fetchUnmaskedCard,
  lockCard,
  toggleDisableCardLimitsTimer,
} from 'models/operations/card-settings/redux/actions';
import { ICardUnmasked, ICard } from 'models/operations/card-settings/entities';
import { IStore } from 'modules/store/types';
import { getCurrentAccountId } from 'models/organizations/redux/selectors';
import UnLockCard from './unlock-process/unlock';
import UnLockConfirmCard from './unlock-process/unlockConfirm';
import DisableCardLimits from './disable-limits-process/disableLimits';
import IncreaseCardLimit from './increase-limit-process/increaseLimit';
import CardLimitsConfirm from './settings-limits-process/settingsLimitsConfirm';

import SettingCardLimits from '../../../components/common/setting-card-limits';

export const CARD_DETAIL_MODAL = 'CARD_DETAIL_MODAL';
export type CardDetailModalType = 'CARD_DETAIL_MODAL';

interface ICardDetailInfoProps {
  fetchCard: (card: string) => void;
  toggleModal: (isShown: boolean) => void;
  blockCard: (cardId: string, accountId: string) => void;
  showUnBlockModal: () => void;
  disableLimitsTimer: () => void;
  unmaskedCard: ICardUnmasked;
  fullCard?: ICard;
  cardId: string;
  accountId: string;
  isLoading: boolean;
  isShownBlock: boolean;
}

export const CardDetailInfo: React.FC<ICardDetailInfoProps> = ({
  fetchCard,
  cardId,
  fullCard,
  isLoading,
  toggleModal,
  showUnBlockModal,
  unmaskedCard,
  accountId,
  blockCard,
  isShownBlock,
  disableLimitsTimer,
}) => {
  useEffect(
    () => {
      fetchCard(cardId);
      disableLimitsTimer();
    },
    [cardId, fetchCard, disableLimitsTimer],
  );

  const { t } = useTranslation();
  const showBlock = () => toggleModal(true);
  const hideBlock = () => toggleModal(false);
  const onBlockCard = () => blockCard(cardId, accountId);
  const status = propOr('', 'status')(fullCard);

  return (
    <CardSettingBox>
      {fullCard && (
        <CardPreview
          cardNumber={fullCard.number}
          expireAt={fullCard.expireAt}
          name={fullCard.owner.embossedName}
          status={fullCard.status}
        />
      )}
      <CardDataPreviewBox>
        <Skeleton isLoading={isLoading} render={<BlockUnmaskedCardSkelet />}>
          <UnmaskedCardPreview card={unmaskedCard} />
        </Skeleton>
        <SettingCardLimits />
      </CardDataPreviewBox>
      <BtnsBox>
        <SpinnerWrapper isLoading={isLoading}>
          {status === AccountStatus.ACTIVE && <BlockCardBtn onClick={showBlock}>{t('block card')}</BlockCardBtn>}
          {status === AccountStatus.LOCKED && (
            <BlockCardBtn onClick={showUnBlockModal}>{t('unblock card')}</BlockCardBtn>
          )}
        </SpinnerWrapper>
      </BtnsBox>
      <PopupBlockCard
        title="Block Card Temporarily"
        description="To block card press the «Block» button. You'll be able to unblock card by yourself."
        actionText="Block"
        onCancel={hideBlock}
        onReject={onBlockCard}
        isShown={isShownBlock}
      />
      <UnLockCard />
      <UnLockConfirmCard />
      <DisableCardLimits />
      <IncreaseCardLimit />
      <CardLimitsConfirm />
    </CardSettingBox>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchCard: (cardId: string) => dispatch(fetchUnmaskedCard(cardId)),
  blockCard: (cardId: string, accountId: string) => dispatch(lockCard(cardId, accountId)),
  toggleModal: (isShown: boolean) => dispatch(toggleBlockCardModal(isShown)),
  showUnBlockModal: () => dispatch(toggleUnBlockCardModal(true)),
  disableLimitsTimer: () => dispatch(toggleDisableCardLimitsTimer(false)),
});

export const mapStateToProps = (store: IStore) => ({
  unmaskedCard: getFullUnmaskedSelectedCard(store),
  fullCard: getSliderActiveCard(store),
  cardId: getSliderActiveCardId(store),
  isShownBlock: getIsShownLockModal(store),
  accountId: getCurrentAccountId(store),
  isLoading: isInitialLoadingCard(store),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CardDetailInfo);
