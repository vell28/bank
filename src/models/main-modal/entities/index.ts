import { AllMainModalContentsType } from 'containers/operations/types';

export type MainModalContentType = AllMainModalContentsType | '';

export interface IMainModalSettings {
  id: MainModalContentType;
  bindUrl: string;
  title?: string;
}

export interface IMainModal {
  isShown: boolean;
  settings: IMainModalSettings;
  step: number;
}
