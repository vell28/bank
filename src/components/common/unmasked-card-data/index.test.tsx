import React from 'react';

import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ICardUnmasked } from 'models/operations/card-settings/entities';
import { UnmaskedCardPreview } from '.';
import { TestWrapper } from '../test-wrapper';

const card: ICardUnmasked = {
  pin: '1234',
  expireAt: '2022-07-01T00:00:00',
  cvc: '000',
  owner: {
    embossedName: 'Card holder',
  },
  number: '5277 6001 0000 4032',
};

describe('<UnmaskedCardPreview /> UI tests', () => {
  test('Should success render <UnmaskedCardPreview />, with pin code', () => {
    const component: any = render(
      <TestWrapper>
        <UnmaskedCardPreview card={card} />
      </TestWrapper>,
    );
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('label').length).toEqual(4);
  });

  test('Test unmask function', () => {
    const component: any = render(
      <TestWrapper>
        <UnmaskedCardPreview card={card} />
      </TestWrapper>,
    );

    // get hidden values
    const hiddenText = component.text();
    expect(hiddenText.indexOf(card.number)).toEqual(-1);
    // TODO: add click function test

    /*    // click on checkbox
    const checkbox = enzymeFind(component, CheckboxBox);
    checkbox.simulate('click');
    // click to switcher
    const switcher = enzymeFind(component, SwitchBox);
    switcher.simulate('click');

    const notHiddenNumber = component.text();
    expect(notHiddenNumber.indexOf(card.number)).not.toEqual(-1); */

    expect(hiddenText).toMatchSnapshot();
  });
});
