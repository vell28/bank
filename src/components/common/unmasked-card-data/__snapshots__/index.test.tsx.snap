// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnmaskedCardPreview /> UI tests Should success render <UnmaskedCardPreview />, with pin code 1`] = `
<div
  class="sc-AxjAm chJldz"
>
  <h2
    class="sc-AxirZ dvplQK"
  >
    Show Card Data
  </h2>
  <div
    class="sc-AxiKw dEvlfm"
  >
    <div
      class="sc-AxmLO ibTasE"
      size="28"
    >
      <span
        class="sc-Axmtr cPocuF"
        size="28"
      />
      <div
        class="sc-fzozJi dDyLif"
      >
        <label
          class="sc-AxhCb iJHZWh"
        >
          <span
            class="sc-AxhUy dSbFOW"
          >
            Card Number
          </span>
          <span
            class="sc-AxgMl Dzbyt"
          >
            5277 60* **** 4032
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw dEvlfm"
  >
    <div
      class="sc-AxmLO ibTasE"
      size="28"
    >
      <span
        class="sc-Axmtr cPocuF"
        size="28"
      />
      <div
        class="sc-fzozJi dDyLif"
      >
        <label
          class="sc-AxhCb iJHZWh"
        >
          <span
            class="sc-AxhUy dSbFOW"
          >
            Expiry Date
          </span>
          <span
            class="sc-AxgMl Dzbyt"
          >
            **/**
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw dEvlfm"
  >
    <div
      class="sc-AxmLO ibTasE"
      size="28"
    >
      <span
        class="sc-Axmtr cPocuF"
        size="28"
      />
      <div
        class="sc-fzozJi dDyLif"
      >
        <label
          class="sc-AxhCb iJHZWh"
        >
          <span
            class="sc-AxhUy dSbFOW"
          >
            CVC
          </span>
          <span
            class="sc-AxgMl Dzbyt"
          >
            ***
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw dEvlfm"
  >
    <div
      class="sc-AxmLO ibTasE"
      size="28"
    >
      <span
        class="sc-Axmtr cPocuF"
        size="28"
      />
      <div
        class="sc-fzozJi dDyLif"
      >
        <label
          class="sc-AxhCb iJHZWh"
        >
          <span
            class="sc-AxhUy dSbFOW"
          >
            PIN
          </span>
          <span
            class="sc-AxgMl Dzbyt"
          >
            ****
          </span>
        </label>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw dEvlfm"
  >
    <div
      class="sc-AxheI ilEEcW"
    >
      <div
        class="sc-fzpans gQNnUB"
      >
        <div
          class="sc-fzplWN bDicWg"
        />
        <span
          class="sc-fzoLsD gJcKWQ"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<UnmaskedCardPreview /> UI tests Test unmask function 1`] = `"Show Card DataCard Number5277 60* **** 4032Expiry Date**/**CVC***PIN****"`;
